pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -264,6 +260,121 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileMachineHealthCheck) remediate(ctx context.Context, needRemediationTargets []target, errList []error, m *mapiv1.MachineHealthCheck) []error {
+	// remediate unhealthy
+	for _, t := range needRemediationTargets {
+		klog.V(3).Infof("Reconciling %s: meet unhealthy criteria, triggers remediation", t.string())
+		if m.Spec.RemediationTemplate != nil {
+			if err := r.externalRemediation(ctx, m, t); err != nil {
+				errList = append(errList, err)
```

Comment:
I would log an error here just like for the internal remediation (maybe add external vs internal to the error message)